source/server/hot_restarting_child.cc

```diff
@@ -0,0 +1,75 @@
+#include "server/hot_restarting_child.h"
+
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Server {
+
+HotRestartingChild::HotRestartingChild(int base_id, int restart_epoch)
+    : HotRestartingBase(base_id), restart_epoch_(restart_epoch) {
+  initDomainSocketAddress(&parent_address_);
+  if (restart_epoch_ != 0) {
+    parent_address_ = createDomainSocketAddress(restart_epoch_ + -1, "parent");
+  }
+  bindDomainSocket(restart_epoch_, "child");
```

Comment:
The one functional change. Since each HotRestarting{Child,Parent} has one of these, and we have one of each, we now have two datagram domain sockets: one for talking to the parent, one for eventually talking to the child. Their sockaddr_un addresses are now differentiated: they include the word "child" if the socket with that address is being used by the child half of the logic, "parent" otherwise.